Source/Initialization/WarpXInitData.cpp

```diff
@@ -869,3 +834,43 @@ void WarpX::CheckGuardCells(amrex::MultiFab const& mf)
         }
     }
 }
+
+void WarpX::InitializeEBGridData(int lev)
+{
+    if(lev==maxLevel()) {
+        if (WarpX::maxwell_solver_id == MaxwellSolverAlgo::Yee ||
+            WarpX::maxwell_solver_id == MaxwellSolverAlgo::CKC ||
+            WarpX::maxwell_solver_id == MaxwellSolverAlgo::ECT) {
+
+            ComputeEdgeLengths();
+            ComputeFaceAreas();
+            ScaleEdges();
+            ScaleAreas();
+
+            const auto &period = Geom(lev).periodicity();
+            WarpXCommUtil::FillBoundary(*m_edge_lengths[lev][0], guard_cells.ng_alloc_EB, period);
+            WarpXCommUtil::FillBoundary(*m_edge_lengths[lev][1], guard_cells.ng_alloc_EB, period);
+            WarpXCommUtil::FillBoundary(*m_edge_lengths[lev][2], guard_cells.ng_alloc_EB, period);
+            WarpXCommUtil::FillBoundary(*m_face_areas[lev][0], guard_cells.ng_alloc_EB, period);
+            WarpXCommUtil::FillBoundary(*m_face_areas[lev][1], guard_cells.ng_alloc_EB, period);
+            WarpXCommUtil::FillBoundary(*m_face_areas[lev][2], guard_cells.ng_alloc_EB, period);
```

Comment:
Do any of these need to be guarded based on dimension? The 0,1,2 makes me think this might misbehave in fewer than 3 dimensions.

Reply:
If these should be made dimension dependent I think that should go in a separate PR since that is different issue.

Reply:
@PhilMiller 
I think it's going to be fine in every case. If you look into warpx.cpp (around line 1610) you'll see that all the three components of these vectors of multifab's are initialized independently of the number of dimensions. Probably the idea is that in fewer dimensionalities memory is not really an issue so it doesn't matter that we initialize some multifabs which are not really needed.

Reply:
As long as it's reasonably small amounts of data, I think that's fine.

Our 2D runs at Modern Electron are limited by memory capacity, though, since we're running a *lot* of particles.

Reply:
I see thanks for the clarification.
Definitely worth a discussion at the next developers meeting :)